### PR TITLE
37.5 Added Architecture

### PR DESCRIPTION
**Pull request by craftamap**

### CODE REVIEW

File: chapters/methodology.tex [hunk 1]
```diff
@@ -6,7 +6,16 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexability
```
Comment (limkokyew) [inline chapters/methodology.tex]: flexability -> flexibility

File: chapters/methodology.tex [hunk 2]
```diff
@@ -6,7 +6,16 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexability
+in switching out various components.
+
+Adding an Face- or Smile-Detector based on the technology of liking requires an
```
Comment (limkokyew) [inline chapters/methodology.tex]: an -> a, I think

File: chapters/methodology.tex [hunk 3]
```diff
@@ -6,7 +6,16 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexability
+in switching out various components.
+
+Adding an Face- or Smile-Detector based on the technology of liking requires an
+developer to implement the intended interface. The implementation then can be
```
Comment (limkokyew) [inline chapters/methodology.tex]: Maybe intended -> designated?

"then can be" should be fine, but I think "can then be" sounds a bit better

File: chapters/methodology.tex [hunk 4]
```diff
@@ -6,7 +6,16 @@ \subsection{User Journey}
...
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexability
+in switching out various components.
+
+Adding an Face- or Smile-Detector based on the technology of liking requires an
+developer to implement the intended interface. The implementation then can be 
+selected by modifing the configuration file of the app. A Factory Method will
```
Comment (limkokyew) [inline chapters/methodology.tex]: modifing -> modifying
"Factory Method" -> "factory method" (?)

**Review by limkokyew — COMMENTED** [review]
Very concise! Found a couple of things you can take a look at.

File: chapters/methodology.tex [hunk 5]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
```
Comment (mac641) [inline chapters/methodology.tex]: We should stay consistent here. 😁
I'd write using no hyphen

File: chapters/methodology.tex [hunk 6]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
...
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying the configuration file of the app. A configuration-based
```
Comment (mac641) [inline chapters/methodology.tex]: I would not use hyphens here as well

File: chapters/methodology.tex [hunk 7]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
...
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying the configuration file of the app. A configuration-based 
+factory will create the Face- or Smile-Detector based on a configuration file. 
+(See \ref{fig:arch1})
+
+The Image Analyzer uses the created implementations of Face- and Smile-
```
Comment (mac641) [inline chapters/methodology.tex]: Not quite sure if this suits our case, but I'd write image analyzer uncapitalised as well 🤔

**Review by mac641 — CHANGES_REQUESTED** [review]
Some consistency issues regarding capitalising certain terms / the use of hyphens -> I'd suggest discussing this on Monday

File: chapters/methodology.tex [hunk 8]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
...
+Detectors. The results of the Image Analyzer are collected and stored in a
+local database.
+
+In order to load and manage the configuration file, an configuration manager
```
Comment (limkokyew) [inline chapters/methodology.tex]: "a configuration [...]" instead of "an configuration [...]"

File: chapters/methodology.tex [hunk 9]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
...
+The Image Analyzer uses the created implementations of Face- and Smile-
+Detectors. The results of the Image Analyzer are collected and stored in a
+local database.
+
+In order to load and manage the configuration file, an configuration manager 
+is used.
+
+\begin{figure}
+    \includegraphics[width=\linewidth]{figures/methodology_architecture_1.jpg}
+    \caption{class diagram describing the architecture}
```
Comment (limkokyew) [inline chapters/methodology.tex]: Capitalize "class"

Something I didn't realize at first as well is that you must briefly describe every figure in your text, so e. g.: "Figure `\ref{fig:arch1}` shows the class diagram of the architecture". You already have "(See `\ref{fig:arch1}`)" above, but you probably have to provide a more detailed description.

**Review by diehmlukas — APPROVED** [review]
I like it

File: chapters/methodology.tex [hunk 10]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
...
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying the configuration file of the app. A configuration-based 
+factory will create the Face- or Smile-Detector based on a configuration file.
```
Comment (limkokyew) [inline chapters/methodology.tex]: You are writing configuration twice: "configuration-based" and "based on a configuration file"

File: chapters/methodology.tex [hunk 11]
```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
```
Comment (AnnaGsell) [inline chapters/methodology.tex:19]: is it really 'switching out'? sounds stange to me, what about 'exchanging'?

File: chapters/methodology.tex (hunk 5 above)
Comment (AnnaGsell) [inline chapters/methodology.tex]: 'liking'?

**Review by AnnaGsell — CHANGES_REQUESTED** [review]

**Review by AnnaGsell — COMMENTED** [review]
Would be  nice to explain model-view-viewmodel in this section...

**Review by diehmlukas — APPROVED** [review]

**Comment by craftamap** [comment]
> Would be nice to explain model-view-viewmodel in this section...

@AnnaGsell done, please rereview!

File: chapters/methodology.tex [hunk 12]
```diff
@@ -14,7 +14,35 @@ \subsection{User Journey} \label{sec:user_journey}
...
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face or Smile Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying a configuration file of the app. A factory will create 
+the Face or Smile Detector based on a configuration file. 
+
+The image analyzer uses the created implementations of Face- and Smile-
```
Comment (mac641) [inline chapters/methodology.tex]: Still `Face-`, `Smile-`...I'd omit the hyphens

**Review by mac641 — APPROVED** [review]
One minor consistency/typo thingy (comment) - other than that, it looks good to me :)
Going to approve it nonetheless 👍🏿

**Review by AnnaGsell — APPROVED** [review]

**Review by limkokyew — APPROVED** [review]

**Pull request merged**